src/test/scala/org/tessellation/consensus/CellMonoidTest.scala

```diff
@@ -0,0 +1,132 @@
+package org.tessellation.consensus
+
+import cats.{Applicative, Monoid, MonoidK}
+import cats.effect.IO
+import org.mockito.{ArgumentMatchersSugar, IdiomaticMockito}
+import org.mockito.cats.IdiomaticMockitoCats
+import org.scalatest.BeforeAndAfter
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+import org.tessellation.Node
+import org.tessellation.consensus.L1ConsensusStep.{L1ConsensusContext, L1ConsensusMetadata}
+import org.tessellation.consensus.transaction.RandomTransactionGenerator
+import org.tessellation.schema.Cell.{NullTerminal, cellMonoid}
+import org.tessellation.schema.{Cell, CellError, StackF, Ω, ΩList}
+import org.tessellation.snapshot.{L0Cell, L0Edge, Snapshot}
+import cats.syntax._
+
+class CellMonoidTest
```

Comment:
Hmm I think we have similar unit tests there:

"Combine pipelines of same types": Lines 95-98: `val combined = M.combine(l1CellA, l1CellB)`
"Combine pipelines of different types": Line 122: `val combined = M.combine(l1Cell, l0Cell)`

>In both cases we want to merge the pipelines by calling combine on their head elements and choosing sequential Cells recursively

Ok I think I get it. So you just want to continuously combine whole stream pipelines instead of testing single combine result 🤔 .

Reply:
Yes exactly. We can just merge two `L1 >>> L2` pipelines

Reply:
Isn't it then a unit test for `Arrow` instead of for `Monoid`? 🤔